src/Parser.zig

```diff
@@ -813,6 +816,27 @@ pub const DeclSpec = struct {
     }
 };
 
+fn typeof(p: *Parser) Error!?Type {
+    if (p.eatToken(.keyword_typeof) orelse p.eatToken(.keyword_typeof1) orelse p.eatToken(.keyword_typeof2)) |_| {
+        const l_paren = try p.expectToken(.l_paren);
+        const start = p.tok_i;
+        if (try p.typeName()) |ty| {
+            try p.expectClosing(l_paren, .r_paren);
+            return ty;
+        } else {
+            p.tok_i = start;
+            if (p.eatToken(.r_paren)) |r_paren| {
+                try p.errTok(.expected_expr, r_paren);
+                return error.ParsingFailed;
+            }
+            const typeof_expr = try p.expr();
```

Comment:
This expression should not be evaluated, see `genericSelection` for how to do that.

The same applies to `sizeof` and `_Alignof` as well so this should probably be extracted into an `exprNoEval` helper.